aicsimageio/readers/tiff_reader.py

```diff
@@ -0,0 +1,287 @@
+                    if (
+                        guessed_dim not in best_guess
+                        and guessed_dim not in dims_from_meta
+                    ):
+                        best_guess.append(guessed_dim)
+                        appended_dim = True
+                        break
+
+                # All of our guess dims were already in the best guess list,
+                # append the dim read from meta
+                if not appended_dim:
+                    best_guess.append(dim_from_meta)
+
+        return "".join(best_guess)
+
+    def _guess_dim_order(self):
+        with self.fs.open(self.path) as open_resource:
+            with TiffFile(open_resource) as tiff:
+                scene = tiff.series[self.current_scene]
+                dims_from_meta = scene.pages.axes
+
+                # If all dims are known, simply return as list
+                if UNKNOWN_DIM_CHAR not in dims_from_meta:
+                    return [d for d in dims_from_meta]
+
+                # Otherwise guess the dimensions and return merge
+                else:
+                    # Get basic guess from shape size
+                    guessed_dims = Reader._guess_dim_order(scene.shape)
+                    return [
+                        d for d in self._merge_dim_guesses(dims_from_meta, guessed_dims)
+                    ]
+
+    @staticmethod
+    def _get_coords(dims: str, shape: Tuple[int]):
+        # Use dims for coord determination
+        coords = {}
+
+        # Use range for channel indices
+        if DimensionNames.Channel in dims:
+            coords[DimensionNames.Channel] = [
+                str(i) for i in range(shape[dims.index(DimensionNames.Channel)])
+            ]
+
+        return coords
+
+    def _read_delayed(self) -> xr.DataArray:
+        """
+        Construct the delayed xarray DataArray object for the image.
+
+        Returns
+        -------
+        image: xr.DataArray
+            The fully constructed and fully delayed image as a DataArray  object.
+            Metadata is attached in some cases as coords, dims, and attrs.
+
+        Raises
+        ------
+        exceptions.UnsupportedFileFormatError: The file could not be read or is not
+            supported.
+        """
+        with self.fs.open(self.path) as open_resource:
+            with TiffFile(open_resource) as tiff:
+                # Get a sample YX plane
+                sample = self._get_image_data(
+                    fs=self.fs,
+                    path=self.path,
+                    scene=self.current_scene,
+                    index=0,
+                )
+
+                # Get shape of current scene
+                # Replace YX dims with empty dimensions
+                operating_shape = tiff.series[self.current_scene].shape
+
+                # If the data is RGB we need to pull in the channels as well
+                if tiff.series[self.current_scene].keyframe.samplesperpixel != 1:
```

Comment:
Here you describe this as channels but in the tests you refer to it as samples.

Reply:
I believe for volumetric data, when each plane is an RGB plane, what we would normally call an RGB channel is an RGB sample. The tests are correct, this comment isn't will update.